dev/release/verify-release-candidate.sh

```diff
@@ -531,16 +531,33 @@ test_integration() {
               $INTEGRATION_TEST_ARGS
 }
 
-clone_testing_repositories() {
-  # Clone testing repositories if not cloned already
-  if [ ! -d "arrow-testing" ]; then
-    git clone https://github.com/apache/arrow-testing.git
+get_or_decompress_source() {
```

Comment:
How about `ensure_source`?

Reply:
Maybe `get_source_distribution` as there is `test_source_distribution`?

Reply:
This may not `get` source distribution. (This may just use a local archive.) So I think that `get` isn't suitable for this case.

Reply:
`ensure_source` suggests verification. Maybe `ensure_source_availability` or something similar.

Reply:
`ensure_source_distribution`? `ensure_source_directory`?